src/libraries/System.Private.CoreLib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -93,7 +93,7 @@ public override unsafe int GetByteCount(char[] chars, int index, int count)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.chars, ExceptionResource.ArgumentOutOfRange_IndexCountBuffer);
             }
 
-            fixed (char* pChars = chars)
+            fixed (char* pChars = &MemoryMarshal.GetArrayDataReference(chars))
```

Comment:
We're trying to reduce the amount of unsafe logic in the core libraries, these types of micro-optimizations aren't really something I think we want to take.

It'd rather be better for the JIT to be able to avoid or elide any additional cost here from the `GetPinnableReference` logic in the case it knows `chars` is non-null.

Reply:
CC. @EgorBo, @jkotas

Reply:
> I do not see a problem with using it in a few more places.

Having said that, I agree that these micro-optimizations add clutter, and the benefit is non-measurable in most cases.

Reply:
Based on the feedback, I’m closing this PR until it’s shown to make a noticeable performance difference.